flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1073,6 +1073,12 @@ bool CompilerInvocation::createFromArgs(
     res.loweringOpts.setLowerToHighLevelFIR(true);
   }
 
+  // -flang-deprecated-no-hlfir
+  if (args.hasArg(clang::driver::options::OPT_flang_deprecated_no_hlfir) &&
+      !args.hasArg(clang::driver::options::OPT_emit_hlfir)) {
```

Comment:
nit: specifying -flang-depricated-nohlfir and -flang-experimental-hlfir together should probably be an error or at least a warning